File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlNegativeValue.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression } from './ISqlExpression';

export class SqlNegativeValue implements ISqlExpression {
    SqlType: SqlType = SqlType.NegativeValue;
    Span: TextSpan = new TextSpan();
    Value!: ISqlExpression;

    Accept(visitor: any): void {
        // 簡單實作
    }

    ToSql(): string {
        return `-${this.Value.ToSql()}`;
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlNegativeValue.cs
namespace T1.SqlSharp.Expressions;

public class SqlNegativeValue : ISqlExpression
{
    public SqlType SqlType { get; set; } = SqlType.NegativeValue;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_NegativeValue(this);
    }

    public required ISqlExpression Value { get; set; }

    public string ToSql()
    {
        return $"-{Value.ToSql()}";
    }
}